zombies.js

```diff
@@ -208,6 +237,136 @@
  * @return {string/boolean}   Weapon name or false if nothing is equipped.
  */
 
+// class Player {
+//   constructor (name, health, strength, speed){
+//     this.name = name;
+//     this.health = health;
+//     this.strength = strength;
+//     this.speed = speed;
+//     this.isAlive = true;
+//     this.equipped = false;
+//     this._pack = [];
+//     this._maxHealth = health;
+//   }
+
+function Player (name, health, strength, speed){
+  this.name = name;
+  this.health = health;
+  this.strength = strength;
+  this.speed = speed;
+  this.isAlive = true;
+  this.equipped = false;
+  let pack = [];
+  this.getPack = function (){
+    return pack;
+  };
+  this.getMaxHealth = function (){
+    return health;
+  };
+}
+
+Player.prototype.checkPack = function (){
+  var packArr = [];
+  for (var i = 0; i < this.getPack().length; i++){
+    packArr.push(this.getPack()[i].name);
+  }
+  console.log(this.name + ' checked the pack. It had\: a ' + packArr.join(", a "));
+};
+
+Player.prototype.takeItem = function (item){
+    if (item instanceof Item === true) {
+      if (this.getPack().length < 3){
```

Comment:
This doesn't handle the case where the player is allowed to store items similar items (items with the same name).